src/test/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,214 @@
+
+import static com.commercetools.sync.producttypes.utils.AttributeDefinitionUpdateActionUtils.buildChangeLabelUpdateAction;
+import static com.commercetools.sync.producttypes.utils.AttributeDefinitionUpdateActionUtils.buildSetInputTipUpdateAction;
+import static com.commercetools.sync.producttypes.utils.AttributeDefinitionUpdateActionUtils.buildChangeIsSearchableUpdateAction;
+import static com.commercetools.sync.producttypes.utils.AttributeDefinitionUpdateActionUtils.buildChangeInputHintUpdateAction;
+import static com.commercetools.sync.producttypes.utils.AttributeDefinitionUpdateActionUtils.buildChangeAttributeConstraintUpdateAction;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AttributeDefinitionUpdateActionUtilsTest {
+    private static AttributeDefinition old;
+    private static AttributeDefinition oldNullValues;
+    private static AttributeDefinitionDraft newSame;
+    private static AttributeDefinitionDraft newDifferent;
+    private static AttributeDefinitionDraft newNullValues;
+
+
+    /**
+     * Initialises test data.
+     */
+    @BeforeClass
+    public static void setup() {
+        old = AttributeDefinitionBuilder
+            .of("attributeName1", LocalizedString.ofEnglish("label1"), StringAttributeType.of())
+            .isRequired(false)
+            .attributeConstraint(AttributeConstraint.NONE)
+            .inputTip(LocalizedString.ofEnglish("inputTip1"))
+            .inputHint(TextInputHint.SINGLE_LINE)
+            .isSearchable(false)
+            .build();
+
+        oldNullValues = AttributeDefinitionBuilder
+            .of("attributeName1", LocalizedString.ofEnglish("label1"), StringAttributeType.of())
+            .isRequired(false)
+            .attributeConstraint(null)
+            .inputTip(null)
+            .inputHint(null)
+            .isSearchable(false)
+            .build();
+
+        newSame = AttributeDefinitionDraftBuilder
+            .of(old)
+            .build();
+
+        newDifferent = AttributeDefinitionDraftBuilder
+            .of(StringAttributeType.of(), "attributeName1", LocalizedString.ofEnglish("label2"), true)
+            .attributeConstraint(AttributeConstraint.SAME_FOR_ALL)
+            .inputTip(LocalizedString.ofEnglish("inputTip2"))
+            .inputHint(TextInputHint.MULTI_LINE)
+            .isSearchable(true)
+            .build();
+
+        newNullValues = AttributeDefinitionDraftBuilder
+            .of(StringAttributeType.of(), "attributeName1", LocalizedString.ofEnglish("label2"), true)
+            .attributeConstraint(null)
+            .inputTip(null)
+            .inputHint(null)
+            .isSearchable(true)
+            .build();
+    }
+
+    @Test
+    public void buildChangeLabelAction_WithDifferentValues_ShouldReturnAction() {
+        final Optional<UpdateAction<ProductType>> result = buildChangeLabelUpdateAction(old, newDifferent);
+
+        assertThat(result).containsInstanceOf(ChangeAttributeDefinitionLabel.class);
```

Comment:
not sure if you need this assert, isn't the following one enough?

Reply:
I saw it in another tests. I agree. I remove it

Reply:
e06fbb36efabf1d88963f9c4f175f76a51c73e36